src/Sarif/Converters/AndroidStudioConverter.cs

```diff
@@ -82,11 +101,13 @@ private void ProcessAndroidStudioLog(XmlReader xmlReader, IResultLogWriter outpu
                 var problem = AndroidStudioProblem.Parse(xmlReader, _strings);
                 if (problem != null)
                 {
-                    output.WriteResult(AndroidStudioConverter.ConvertProblemToSarifIssue(problem));
+                    results.Add(ConvertProblemToSarifIssue(problem));
```

Comment:
Oh, I see. You are collecting all the results but are still piping them to the writer one at a time. Maybe we should support a second WriteResults helper to provide minor assistance with this pattern.